unit/goto-programs/goto_program_validate.cpp

```diff
@@ -0,0 +1,429 @@
+/*******************************************************************\
+
+ Module: Unit tests for goto program validation
+
+ Author: Diffblue Ltd.
+
+ Date: Oct 2018
+
+\*******************************************************************/
+
+/// \file
+/// Unit tests for goto program validation
+
+#include <testing-utils/catch.hpp>
+#include <testing-utils/message.h>
+
+#include <goto-programs/goto_convert_functions.h>
+#include <goto-programs/validate_goto_model.h>
+
+#include <util/arith_tools.h>
+#include <util/c_types.h>
+#include <util/std_code.h>
+#include <util/validate.h>
+
+namespace
+{
+code_function_callt make_void_call(const symbol_exprt &function)
+{
+  code_function_callt ret;
+  ret.function() = function;
+  return ret;
+}
+
+void validate(
+  const goto_functionst &goto_functions,
+  const validation_modet vm,
+  const goto_model_validation_optionst &goto_model_validation_options)
+{
+  validate_goto_model(goto_functions, vm, goto_model_validation_options);
+}
+
+bool test_for_pass(
+  goto_modelt &goto_model,
+  const goto_model_validation_optionst goto_model_validation_options)
+{
+  validate(
+    goto_model.goto_functions,
+    validation_modet::INVARIANT,
+    goto_model_validation_options);
+  return true;
+}
+
+bool test_for_failure(
```

Comment:
Rather than doing this `test_for_failure` function, can you just use the built in `REQUIRE_THROWS_AS` macro whereever you otherwise use this `test_for_failure` function? e.g. this model is used in `unit/util/symbol_table.cpp`

Reply:
Done - but have kept a wrapper to keep the boilerplate to a minimum. Let me know.